rs/cli/src/registry_dump.rs

```diff
@@ -0,0 +1,161 @@
+use std::{collections::BTreeMap, fs, path::PathBuf, str::FromStr, time::Duration};
+
+use ic_base_types::RegistryVersion;
+use ic_registry_keys::{
+    DATA_CENTER_KEY_PREFIX, NODE_OPERATOR_RECORD_KEY_PREFIX, NODE_RECORD_KEY_PREFIX, SUBNET_RECORD_KEY_PREFIX,
+};
+
+use ic_protobuf::registry::{
+    dc::v1::DataCenterRecord, node::v1::NodeRecord, node_operator::v1::NodeOperatorRecord, subnet::v1::SubnetRecord,
+};
+
+use anyhow::Error;
+use ic_interfaces_registry::{RegistryClient, RegistryValue};
+use ic_management_types::Network;
+use ic_registry_local_registry::LocalRegistry;
+use service_discovery::registry_sync::sync_local_registry;
+use slog::Logger;
+use uuid::Uuid;
+
+pub async fn dump_registry(
+    logger: Logger,
+    path: &Option<PathBuf>,
+    version: &i64,
+    network: &Network,
+) -> Result<(), Error> {
+    let (path, should_dispose) = match path {
+        Some(val) => (val.clone(), false),
+        None => {
+            let uuid = Uuid::new_v4();
+            let binding = format!("~/tmp/{}", uuid);
+            let local_temp = shellexpand::tilde(&binding);
+            (
+                PathBuf::from_str(&local_temp).map_err(|e| anyhow::anyhow!("Error converting to path: {:?}", e))?,
+                true,
+            )
+        }
+    };
+
+    sync_local_registry(logger.clone(), path.clone(), vec![network.get_url()], false, None, None)
```

Comment:
We already have a similar function in the dashboard backend, and it's accessible from the cli as well. Is there a good reason why we need to have both?

Reply:
I didn't see that backand was a lib as well. I will integrate that! Thanks!